src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,219 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.InvalidInputException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static final String STANDARD_LOGIN_OK = "Nick";
+    private static final String PASSWORD_OK = "adgfaw123";
+    private static final long STANDARD_ID_OK = 633444;
+    private static final int STANDARD_AGE_OK = 18;
+    private static final String SHORT_LOGIN = "Hi";
+    private static final String SHORT_PASSWORD = "5";
+    private static final int AGE_LESS = 15;
+    private static final int AGE_BIGGEST = 165;
+    private static final long NEGATIVE_ID = -4324;
+    private static final String EMPTY = "";
+    private static final String SPECIAL_CHARACTERS = "@#$%";
+    private static final String LARGE_LOGIN = "HelloMyNameIsJohnPhilips";
+    private static final String LARGE_PASSWORD = "fesgeFEFr32453ft4t3t43534543";
+    private static RegistrationService registrationService;
+    private static User standartUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        standartUser = userConstructor(STANDARD_LOGIN_OK,
+                PASSWORD_OK,
+                STANDARD_AGE_OK,
+                STANDARD_ID_OK);
+    }
+
+    @AfterEach
+    void clear() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_duplicateUser_notOk() {
+        User secondUser = userConstructor(STANDARD_LOGIN_OK,
+                PASSWORD_OK,
+                STANDARD_AGE_OK,
+                STANDARD_ID_OK);
+        Storage.people.add(standartUser);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(secondUser),
+                "User already exist");
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        standartUser = null;
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "You can't register null user");
+    }
+
+    @Test
+    void register_user_specialCharacterLogin_notOk() {
+        standartUser.setLogin(SPECIAL_CHARACTERS);
+        assertThrows(InvalidInputException.class,
+                () -> registrationService.register(standartUser),
+                "You can't register login with special characters, use only letters and numbers");
+    }
+
+    @Test
+    void register_user_NullLogin_notOk() {
```

Comment:
```suggestion
    void register_nullUserLogin_notOk() {
```
Let's follow the same naming convention